### PR TITLE
Initial source commit

### PR DESCRIPTION
**Pull request by amansrivastava17**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,170 @@
-# spello
\ No newline at end of file
+<p align="center">
+    <img src="logo.png" width="350", height="220">
```
Comment (krupalmodi18) [inline README.md:2]: Change text in logo to "A Fast and accurate SpellCorrector"

File: README.md [hunk 2]
```diff
@@ -1 +1,170 @@
...
+<p align="center">  
+ <a href="#what-is-it">What is it</a> •  
+  <a href="#-installation">Installation</a> •  
+  <a href="#-️getting-started">Getting Started</a> 
+</p>  
+</p>
+
+<h2 align="center">What is it</h3>  
+  
+**Spello** is a spellcorrection model using power of two model Symspell and Phoneme in backend to get the best possible spelling suggestion for misspelled words in a text. Phoneme Model uses <a href="https://en.wikipedia.org/wiki/Soundex">Soundex</a> algo in background and is responsible to suggest corrections for sound related mistakes in word. Also, we are using a modifield version of<a href="https://github.com/wolfgarbe/SymSpell"> Symspell</a> model to get spell suggestions based on edit-distances. <br>
```
Comment (krupalmodi18) [inline README.md]: "Spello is a spellcorrection model built with combination of two models, Phoneme (link should be there) and Symspell (link should be there). Phoeme Model uses Soundex algo in background and suggests correct spellings using phonetic concepts to identify similar sounding words. On the other hand, Symspell Model uses concept of edit-distance in order to suggest correct spellings. Spello (link to our blog) get's you best of both, taking into consideration context of the word as well.

File: README.md [hunk 3]
```diff
@@ -1 +1,170 @@
...
+Currently, this module is available for **English(en)** and 10 other indian languages which are **Hindi(hi)**,  **Marathi(mr)**,  **Bengali(bn)**,  **Punjabi(pa)**, **Gujarati(gu)**, **Oriya(or)**, **Tamil(ta)**, **Telegu(tl)**, **Kannada(kn)**, **Malyalam(ml)**.
```
Comment (krupalmodi18) [inline README.md]: We have not tested it for other languages and we don't know how it works. So we just say english and Hindi. We are not giving pretrained models for other languages also.

File: README.md [hunk 4]
```diff
@@ -1 +1,170 @@
...
+```  
+
+#### 2. Model Training
+Here, you can provide two types of training data to the model
+- List of text or sentences.
+- Dict having word and their corresponding count.
+
+**Training providing list of sentences**
+```python 
+>>> sp.train(['my name is aman', 'this is a text corpus'])
```
Comment (krupalmodi18) [inline README.md:68]: We should change example. Changing Naman to Aman is not the right thing.

File: README.md [hunk 5]
```diff
@@ -1 +1,170 @@
...
+Currently, this module is available for **English(en)** and 10 other indian languages which are **Hindi(hi)**,  **Marathi(mr)**,  **Bengali(bn)**,  **Punjabi(pa)**, **Gujarati(gu)**, **Oriya(or)**, **Tamil(ta)**, **Telegu(tl)**, **Kannada(kn)**, **Malyalam(ml)**.  
+  
+<h2 align="center">💾 Installation</h2>  
+<p align="right"><a href="#what-is-it"><sup>▴ Back to top</sup></a></p>
+Install the spello via `pip`
+
+```bash  
+$ pip install spello
+```  
+
```
Comment (krupalmodi18) [inline README.md:50]: Add "You can either train a new model from scratch or use pre-trained model. Alternatively you can also train model for your domain and use that on priority while use pre-trained model as a fallback."

File: README.md [hunk 6]
```diff
@@ -1 +1,170 @@
...
+<p align="right"><a href="#what-is-it"><sup>▴ Back to top</sup></a></p>
+  
+#### 1. **Model Initialisation**
+Initialise the model for one of the suppored languages. 
+```python  
+>>> from spello.model import SpellCorrectionModel  
+>>> sp = SpellCorrectionModel(language='en')  
+```  
+
+#### 2. Model Training
```
Comment (krupalmodi18) [inline README.md]: "Model training - New model"

File: README.md [hunk 7]
```diff
@@ -1 +1,170 @@
...
+Here, you can provide two types of training data to the model
```
Comment (krupalmodi18) [inline README.md]: "You can choose to train model by providing data in one of the following format"

File: README.md [hunk 8]
```diff
@@ -1 +1,170 @@
...
+
+#### 5. Load Model 
+Load the trained model from saved path, First initialise the model and call the load method
+```python  
+>>> from spello.model import SpellCorrectionModel
+>>> sp = SpellCorrectionModel(language='en')
+>>> sp.load('/home/ubuntu/model.pkl')
+```  
+
+#### 6. Customize Configuration of Model
```
Comment (krupalmodi18) [inline README.md]: Add (Optional)

File: README.md [hunk 9]
```diff
@@ -1 +1,170 @@
...
+Here, you are also provided to customize various configuration of the model like
```
Comment (krupalmodi18) [inline README.md:101]: "You can choose to customize various following configurations of the model"

File: README.md [hunk 10]
```diff
@@ -1 +1,170 @@
...
+>>> sp.spell_correct('i wnt to ply futbal')
+{'original_text': 'i wnt to ply futbal',
+ 'spell_corrected_text': 'i want to play football',
+ 'correction_dict': {'wnt': 'want', 'ply': 'play', 'futbal': 'football'}
+}
+
+```
+
+To train model for other languages, you can download data from <a href="https://wortschatz.uni-leipzig.de/en/download/">here </a> and follow training process.
+
```
Comment (krupalmodi18) [inline README.md:144]: Add section of limitations and contribution guidelines.

File: README.md [hunk 11]
```diff
@@ -1 +1,170 @@
...
+```
+*To reset to default config*
+```python
+>>> sp.set_default_config()
+```
+there are many more configurations which you can set, check this <a href="https://github.com/hellohaptik/spello/blob/master/spello/config.py">file</a> for more details
+
+
+## Get Started with Pre-trained Models
+Here are few simple models. They are trained on 30K news + 30k wikipedia sentences.
```
Comment (krupalmodi18) [inline README.md]: "We have trained a basic model on on 30K news + 30k wikipedia sentences". Avoid starting sentences form "Here..."

File: README.md (hunk 1 above)
Comment (amansrivastava17) [reply README.md:2]: Done

File: README.md (hunk 2 above)
Comment (amansrivastava17) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (amansrivastava17) [reply README.md]: Sure

File: README.md (hunk 4 above)
Comment (amansrivastava17) [reply README.md:68]: Done

File: README.md (hunk 5 above)
Comment (amansrivastava17) [reply README.md:50]: DOne

File: README.md (hunk 7 above)
Comment (amansrivastava17) [reply README.md]: Done

File: README.md (hunk 9 above)
Comment (amansrivastava17) [reply README.md:101]: Done

File: README.md (hunk 11 above)
Comment (amansrivastava17) [reply README.md]: Done

**Pull request merged**